traits/observers/tests/test_trait_added_observer.py

```diff
@@ -0,0 +1,288 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+from unittest import mock
+
+from traits.has_traits import HasTraits
+from traits.trait_types import Str
+from traits.observers._trait_added_observer import (
+    TraitAddedObserver,
+    _RestrictedNamedTraitObserver,
+)
+from traits.observers._testing import (
+    call_add_or_remove_notifiers,
+    create_graph,
+    DummyNotifier,
+    DummyObserver,
+)
+
+
+def create_observer(**kwargs):
+    values = dict(
+        match_func=mock.Mock(),
+        optional=False,
+    )
+    values.update(kwargs)
+    return TraitAddedObserver(**values)
+
+
+class DummyMatchFunc:
+    """ A callable to be used as TraitAddedObserver.match_func
+    """
+
+    def __init__(self, return_value):
+        self.return_value = return_value
+
+    def __call__(self, name, trait):
+        return self.return_value
+
+    def __eq__(self, other):
+        return self.return_value == other.return_value
+
+    def __hash__(self):
+        return hash(self.return_value)
+
+
+class TestTraitAddedObserverEqualHashImmutable(unittest.TestCase):
```

Comment:
That's an impressive test class name ...

Reply:
Hmmm the "Immutable" part needs to go, removing...